akka-http-core/src/main/scala/akka/http/impl/engine/client/HttpsProxyGraphStage.scala

```diff
@@ -48,7 +49,13 @@ private final class HttpsProxyGraphStage(targetHostName: String, targetPort: Int
 
   override def shape: BidiShape[ByteString, ByteString, ByteString, ByteString] = BidiShape.apply(sslIn, bytesOut, bytesIn, sslOut)
 
-  private val connectMsg = ByteString(s"CONNECT $targetHostName:$targetPort HTTP/1.1\r\nHost: $targetHostName\r\n\r\n")
+  private def createProxyHeader() = {
+    proxyAuth.fold("") { httpCredentials ⇒
+      s"${`Proxy-Authorization`(httpCredentials)}\r\n"
```

Comment:
We rely on the `toString` of `Proxy-Authorization` here - is that OK? No encoding/escaping needed for example? Is there a 'more explicit' method available rather than relying on the toString?